src/bot.py

```diff
@@ -359,6 +364,7 @@ def on_end_game(self):
     def on_end_run(self):
         success = self._char.tp_town()
         self._tps_left -= 1
+        self._pre_buffed = 0
```

Comment:
this shouljd be in end_game(), no? end_run() is called after pindle and shenk routine.